controlplane/kubeadm/internal/etcd/etcd.go

```diff
@@ -151,7 +151,12 @@ func NewClient(ctx context.Context, config ClientConfiguration) (*Client, error)
 		return nil, errors.Wrap(err, "unable to create etcd client")
 	}
 
-	return newEtcdClient(ctx, etcdClient)
+	client, err := newEtcdClient(ctx, etcdClient)
+	if err != nil {
+		closeErr := etcdClient.Close()
+		return nil, errors.Wrap(err, closeErr.Error())
+	}
+	return client, err
 }
 
 func newEtcdClient(ctx context.Context, etcdClient etcd) (*Client, error) {
```

Comment:
Can you please update l.170 to:
```go
		return nil, errors.Wrap(err, "failed to get etcd status")
```
(Just to make sure we get unique identifiable errors in all cases in l.157 which makes debugging a lot easier)

Reply:
Will do!